### PR TITLE
Polyline ontap improvements

### LINKED ISSUE #14: Multiple onTap called when polylines are close to each other
If there are many polylines multiple onTap can be called. A solution where only the closest one will be triggered should be implemented. Will see what i can do about it later.

### PR DESCRIPTION
**Pull request by Mkohm**
Close #14

### CODE REVIEW

**Comment by HugoHeneault** [comment]
Hey @Mkohm 

Thanks for this PR. 👍 

It seems that you also commited lots of code reformating. Would you please only commit the *real* changes you made so we keep the formating this way?

We're using dartfmt 

Thanks !

**Comment by Mkohm** [comment]
ready for merge ?

**Comment by tuarrep** [comment]
Waiting #17

File: lib/flutter_map_tappable_polyline.dart [hunk 1]
```diff
@@ -208,7 +149,86 @@ class TappablePolylineLayer extends StatelessWidget {
...
+        // To determine if we have tapped in between two po ints, we
+        // calculate the length from the tapped point to the line
+        // created by point1, point2. If this distance is shorter
+        // than the specified threshold, we have detected a tap
+        // between two points.
+        //
+        // We start by calculating the length of all the sides using pythagoras.
+        var a = _distance(point1, point2);
+        var b = _distance(point1, tap);
+        var c = _distance(point2, tap);
```
Comment (tuarrep) [inline lib/flutter_map_tappable_polyline.dart:179]: We are introducing breaking change here by privatizing  `distance`. As we’ve released v1 we could not afford that FTM.

Could you please:
- add a `distance` method that proxies call to `_distance`;
- mark it as deprecated;
- add a TODO to remind us to delete it in V2

**Review by tuarrep — CHANGES_REQUESTED** [review]
LGTM except the introduction of a breaking change.

**Comment by tuarrep** [comment]
@HugoHeneault could you double check this since formatting and rebasing was done?

File: lib/flutter_map_tappable_polyline.dart (hunk 1 above)
Comment (Mkohm) [reply lib/flutter_map_tappable_polyline.dart:179]: Done!

Comment (tuarrep) [reply lib/flutter_map_tappable_polyline.dart:179]: Great

**Review by tuarrep — APPROVED** [review]

**Comment by HugoHeneault** [comment]
Thanks guys for the good work/review. I won't have time to review it this week, probably on monday or tuesday. :-)

**Comment by HugoHeneault** [comment]
Thanks a lot for this! 👏

**Pull request merged**